### PR TITLE
[ROCm] Add AMDSMI support for UUID input

### PR DESCRIPTION
**Pull request by jataylo**
Adds support for for using UUIDs for AMDSMI utilities in PyTorch via CUDA_VISIBLE_DEVICES/HIP_VISIBLE_DEVICES.

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @hongxiayang @naromero77amd

### CODE REVIEW

File: test/test_cuda.py [hunk 1]
```diff
@@ -4426,25 +4426,91 @@ def free():
...
         self.assertTrue(torch.cuda.power_draw() >= 0)
 
-    @unittest.skipIf(TEST_PYNVML, "pynvml is not available")
+    @unittest.skipIf(TEST_PYNVML, "pynvml/amdsmi is not available")
     def test_clock_speed(self):
         self.assertTrue(torch.cuda.clock_rate() >= 0)
 
+    @unittest.skipIf(TEST_PYNVML, "pynvml/amdsmi is not available")
+    @unittest.skipIf(not TEST_WITH_ROCM, "amdsmi specific test")
+    def test_raw_amdsmi_device_count(self):
```
Comment (jithunnair-amd) [inline test/test_cuda.py:4450]: I'd suggest adding comment before each of these new tests to explain what the test is trying to do.

File: torch/cuda/__init__.py [hunk 2]
```diff
@@ -743,11 +743,15 @@ def _raw_device_uuid_amdsmi() -> Optional[List[str]]:
             warnings.warn("Cannot get amd device handler")
             return None
         try:
-            uuid = amdsmi.amdsmi_get_gpu_device_uuid(handler)
+            uuid = amdsmi.amdsmi_get_gpu_asic_info(handler)["asic_serial"][
```
[the remark this answers is not in the document]
Comment (jataylo) [reply torch/cuda/__init__.py:746]: Nothing in the changelog indicates that the API has changed between 6.0-6.2 
https://github.com/ROCm/amdsmi/blob/develop/CHANGELOG.md

**Comment by jataylo** [comment]
cc: @malfet @xw285cornell

**Comment by pruthvistony** [comment]
@malfet ,
Please review this PR.

**Review by pruthvistony — APPROVED** [review]

**Comment by jataylo** [comment]
Hey @malfet @xw285cornell any comments on this one?

**Review by jeffdaily — APPROVED** [review]

**Pull request closed without merging**